File: 은지/LV2/타겟 넘버.js
function solution(numbers, target) {
  var answer = 0
  add(0, 0, numbers, target)
  function add(i, sumV, n, t) {
    if (i === n.length) {
      if (sumV === t) {
        answer += 1
      }      
    } else {
      add(i+1, sumV+n[i], n, t)
      add(i+1, sumV-n[i], n, t)
    }
  }
  return answer;
}

numbers = [1, 1, 1, 1, 1]
target = 3
console.log(solution(numbers, target))

File: 은지/LV2/타겟 넘버.py
answer = 0

def add(i, sumV, numbers, target):
    global answer

    if i == len(numbers):
        if sumV == target:
            answer += 1
    else:
        add(i+1, sumV+numbers[i], numbers, target)
        add(i+1, sumV-numbers[i], numbers, target)


def solution(numbers, target):
    add(0, 0, numbers, target)
    return answer

numbers = [1, 1, 1, 1, 1]
target = 3
print(solution(numbers, target))